exp/ipam/api/v1beta1/condition_consts.go

```diff
@@ -0,0 +1,13 @@
+package v1beta1
+
+// These reasons are intended to be used with failed Ready conditions on an [IPAddressClaim].
+const (
+	// AllocationFailedReason is the reason used when allocating an IP address for a claim fails. More details should be
+	// provided in the condition's message.
+	// When the IP pool is full, [PoolExhaustedReason] should be used for better visibility instead.
+	AllocationFailedReason = "AllocationFailed"
```

Comment:
Question here is whether it makes sense to have more specific reasons depending on provider, and whether this one is therefore superfluous. In general it probably doesn't matter why the allocation failed exactly, and details can still be provided in the condition message.
The only exception is the pool being full, which can be remediated, which is why the second reason below exists.

Reply:
I think it's fine to add this reason. It should be good enough for most cases and if someone wants to have a more specific infra-provider reason they can still use it

Reply:
>The only exception is the pool being full, which can be remediated, which is why the second reason below exists.

This makes me think if AllocationFailed should be a condition on its own and PoolNotReady a reason. But you're more familiar than me so feel free to disregard.

Reply:
Yeah, I thought about that as well. Since there is only one thing that happens - you either get an address or you don't - I don't really think it makes sense to have multiple conditions.